### ISSUE TITLE
Extensive solid infill causing printer nozzle hits

### ISSUE THREAD

**Issue by TommyC81** [supermerill/SuperSlicer#1090]
**Describe the bug**
Not sure if bug or intended, however, SuperSlicer creates an extensive amount of solid infill that (due to other printer settings or otherwise) sets the scene for extensive situations where the printer nozzle is likely to hit the infill (as experienced by myself). Not sure if setting missed somewhere, but I'm unable to reduce the amount of solid infill. The same does not occur in Cura (Arachne beta).

**To Reproduce**
Attached projects and gcode from both SuperSlicer and Cura (Arachne beta), check files to note infill. Additional images below.

**3D Model and Slic3r Configuration Export**
[SuperSlicer - Penholder.zip](https://github.com/supermerill/SuperSlicer/files/6328273/SuperSlicer.-.Penholder.zip)
[SuperSlicer - Penholder.gcode.zip](https://github.com/supermerill/SuperSlicer/files/6328277/SuperSlicer.-.Penholder.gcode.zip)

[Cura (Arachne) - Penholder.zip](https://github.com/supermerill/SuperSlicer/files/6328274/Cura.Arachne.-.Penholder.zip)
[Cura (Arachne) - Penholder.gcode.zip](https://github.com/supermerill/SuperSlicer/files/6328278/Cura.Arachne.-.Penholder.gcode.zip)

**Expected behavior**
Less solid infill and better use of thicker lines where appropriate (thus also reducing the the chance of over extrusion and other issues).

**Screenshots**
SuperSlicer:
![SuperSlicer - Layer 513](https://user-images.githubusercontent.com/1904092/115094049-4121a200-9f2d-11eb-8ba9-7db57dae033a.PNG)

Cura (Arachne beta):
![Cura (Arachne) - Layer 513](https://user-images.githubusercontent.com/1904092/115094054-48e14680-9f2d-11eb-9d41-d35856543d71.PNG)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Version: 20H2

**Additional context**

**Comment by TommyC81** [comment]
Is this possibly related to #253 and https://github.com/slic3r/Slic3r/issues/5066?

**Comment by jesterhead82** [comment]
Would it be possible to define a different infill type for solid infill which got generated by the slicer to ensure hor./vert. thickness/unfilled shells?

e.g. with concentric infill the issue is less intrusive (less wobble, less overextrusion of material).

Solid infill Rect:
![image](https://user-images.githubusercontent.com/11690785/116620568-4a096f00-a942-11eb-90fc-2480aa213331.png)

Solid infill concentric:
![image](https://user-images.githubusercontent.com/11690785/116620725-82a94880-a942-11eb-8914-99c06ae787c2.png)

**Comment by neophyl** [comment]
I agree this is a problem but if all you do is replace it with concentric then you may as well just increase the perimeter count.  Which often removes the solid infill anyway.

**Comment by supermerill** [comment]
> e.g. with concentric infill the issue is less intrusive (less wobble, less overextrusion of material).

you can also try 'rectilinear w gapfill'

I won't work on it right now, have to think of a way to "fix" it. Maybe add a setting for bottom layer from the build plate only?

**Comment by OldCurmudgeon3DP** [comment]
I just noticed (while researching a FB post on this issue) that the slicer seems to still be enforcing some form of the Ensure Vertical Shell Thickness option even when unchecked. I created a cylinder and rotated it 60deg from vertical. The solid infill pattern appeared on "overhang" colored areas until I increased the perimeter count to 7. Below 6, the area circled in green begins to grow. I have the shell thickness option unchecked. When checked, gap fill areas are mixed in with the solid infill even at 7 perimeters.

<img width="551" alt="Screenshot 2021-07-26 132826" src="https://user-images.githubusercontent.com/44645861/127032758-edf9452b-cb6f-4543-83ea-adeeef07ad89.png">

<img width="550" alt="Screenshot 2021-07-26 132652" src="https://user-images.githubusercontent.com/44645861/127032742-80923f2a-74f5-4ea2-b27d-05a5199b33af.png">

<img width="644" alt="Screenshot 2021-07-26 133524" src="https://user-images.githubusercontent.com/44645861/127033434-5cc9ce07-adb3-4d91-b4c3-1132da5c74b1.png">

**Comment by Zeblote** [comment]
> Solid infill concentric:
> ![image](https://user-images.githubusercontent.com/11690785/116620725-82a94880-a942-11eb-8914-99c06ae787c2.png)

@jesterhead82 how did you get it to do that?

**Comment by Zeblote** [comment]
> but if all you do is replace it with concentric then you may as well just increase the perimeter count. Which often removes the solid infill anyway.

@neophyl no, that's not it. The idea is to only to this for slopes, not the entire part. Currently that is only possible with slow wiggle infill.

**Comment by neophyl** [comment]
I think you missed my point.  Yes I am familiar with the solid infill (what you call the wiggle infill) being added in an undesirable manner on sloping surfaces.  Its been raised as an issue for quite awhile on the Prusa slicer github as well as several times here.   I agree this behavior should really be changed.  I am not saying it doesn't need looking at.   The problem is that from reading both here and on the prusa git that it doesnt seem like a simple fix.  Its one of those ones that has been around for ages.

The point I was making is that if you change your slicer settings to replace it with concentric infill to try and get rid of it then from a practical standpoint theres not much difference with extra perimeters.  Seeing as in SOME case increasing perimeter count can reduce or remove the unwanted infill it seems like a first option to try when slicing to possibly get a better result.
